### PR TITLE
Combobox: Documentation

### LINKED ISSUE #93036: New Select: Document component
- Write documentation in Storybook
- Decide if documentation is required for the Saga website too
- Add information about `comboboxTestSetup` helper, and mocking getBoundingClientRect

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

Docs!

**Why do we need this feature?**

So people understand stuff!

**Who is this feature for?**
Devs

**Which issue(s) does this PR fix?**:

Fixes [#93036](https://github.com/grafana/grafana/issues/93036)

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Combobox/Combobox.mdx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+**Do**
+
+- Use in inline query editors
+- Use when you require async calls from a select input
+
+**Don't**
+
+- Use the async functionality, when all items are only loaded on the initial load
+- Use when fewer than 4 items are needed, as a `RadioButtonGroup` may be more suitable (not for inline use cases)
+- Use this component if you need custom components as options
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Combobox/Combobox.mdx:18]:
```suggestion
- Use this component if you need custom option styling
```

File: packages/grafana-ui/src/components/Combobox/Combobox.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+The recommended way to set the width is by sizing the container element. This is so it may reflect a similar size as other inputs in the context.
+
+If that is not possible, the width can be set directly on the component, by setting a number, which is a multiple of `8px`.
+
+For inline usage, such as in query editors, it may be useful to size the input based on the content. Set `width="auto"` to achieve this. In this case, it is also recommended to set `maxWidth` and `minWidth`.
+
+## Async Usage
+
+The `options` prop can accept and async function:
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Combobox/Combobox.mdx:40]:
```suggestion
The `options` prop can accept an async function:
```

File: packages/grafana-ui/src/components/Combobox/Combobox.mdx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  Object.defineProperty(Element.prototype, 'getBoundingClientRect', {
+    value: mockGetBoundingClientRect,
+  });
+});
+```
+
+### Selecting an option
+
+To select an option, you can any `*ByRole` methods, as Combobox has proper roles for accessibility.
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Combobox/Combobox.mdx:72]:
```suggestion
To select an option, you can use any `*ByRole` methods, as Combobox has proper roles for accessibility.
```

**Review by ashharrison90 — APPROVED** [review]
just spelling nits really 😄 looks great!

File: packages/grafana-ui/src/components/Combobox/Combobox.mdx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+## Async Usage
+
+The `options` prop can accept an async function:
+
+- When the menu opens, the `options` function is called with `''`, to load all options.
+- When the user types, the `options` function is called with the current input value.
+
+Note: The calls are debounced. Old calls are invalidated when a new call is made.
+
+## Unit testing
```
Comment (eledobleefe) [inline packages/grafana-ui/src/components/Combobox/Combobox.mdx:47]: Love you added also info about this 🤗

**Review by eledobleefe — APPROVED** [review]

**Pull request merged**